### PR TITLE
Add ST7567 display support

### PR DESCRIPTION
**Pull request by condemil**
Relates to #1451

### CODE REVIEW

File: config/example-extras.cfg [hunk 1]
```diff
@@ -1221,9 +1221,14 @@
...
-#contrast: 40
+#rs_pin:
+#cs_pin:
+#a0_pin:
+#   The pins connected to an st7567 type lcd. These parameters must be
+#   provided when using an st7567 display.
+#contrast: 50
 #   The contrast to set when using a uc1701 type display.  The value may
-#   range from 0 to 63.  Default is 40.
+#   range from 0 to 63.  Default is 50.
```
Comment (condemil) [inline config/example-extras.cfg]: It is not possible to see any text with default contrast 40 and can be misleading during configuration.

Comment (Arksine) [reply config/example-extras.cfg]: It depends on the display.  The display that I developed UC1701 support with shows well at 40, not great anywhere else.  If ST7567 needs a different default value then I recommend making it different.  By making this change you will affect working UC1701 configurations.

Comment (condemil) [reply config/example-extras.cfg]: If UC1701 shows well at 40 then I agree with different default values per display type. I made it different for ST7567.

**Comment by KevinOConnor** [comment]
Thanks.

Some comments:
- the commit will require a signed-off-by line - see: https://github.com/KevinOConnor/klipper/blob/master/docs/CONTRIBUTING.md
- if a change is needed to the uc1701 behaviour then it should be done in a separate commit.  (Though, as @Arksine points out, it's not clear if that change should be made.)
- Please don't claim a copyright to display.py as I don't think the change is significant.  (I'd prefer to avoid the copyrights becoming a "laundry list" of everyone that's touched a file.)

-Kevin

**Comment by condemil** [comment]
I updated the commit message and removed copyright from display.py. Change of uc1701 behavior was removed.

**Comment by KevinOConnor** [comment]
Thanks.

-Kevin

**Comment by KevinOConnor** [comment]
FYI, I am thinking of removing support for `lcd_type: st7567` as testers are reporting that the more common `lcd_type: uc1701` works on that display.

-Kevin

**Pull request merged**